pkg/secretless/plugin/sharedobj/external_plugins.go

```diff
@@ -136,43 +136,44 @@ func ExternalPlugins(
 	for rawPluginName, rawPlugin := range rawPlugins {
 		logger.Infof("Loading plugin '%s'...", rawPluginName)
 
+		pluginLogger := logger.CopyWith(rawPluginName, logger.DebugEnabled())
```

Comment:
So this is a pretty serious code smell with our design.  Can you give me a little more context about the reason for this?  The whole point of injecting a logger is so that the code using it only has to call logging methods.  It should have _no knowledge_ of how the logger is configured. 

To this another way: These are error happening inside the `ExternalPlugins` func.  But we're making them _look_ like errors from inside of something else. 

I suspect that there's a deeper design issue here that you're hacking around to solve, but I need a bit more context to tell what it is.

Reply:
@jonahx I'm just using this other logger to prepend the errors with the plugin name

Reply:
@doodlesbykumbi So this is not how prefixes are intended to be used.  As I said, this makes it appear that the code is _coming from within_ the particular plugins.  In this case, it looks like you want to use simple error wrapping.

There are [some new facillities for doing this](https://blog.golang.org/go1.13-errors):

> In Go 1.13, the fmt.Errorf function supports a new %w verb. When this verb is present, the error returned by fmt.Errorf will have an Unwrap method returning the argument of %w, which must be an error. In all other ways, %w is identical to %v.
> ```go 
> if err != nil {
>     // Return an error which unwraps to err.
>     return fmt.Errorf("decompress %v: %w", name, err)
> }
> ```